libs/dyn/merge/override.go

```diff
@@ -0,0 +1,204 @@
+package merge
+
+import (
+	"fmt"
+
+	"github.com/databricks/cli/libs/dyn"
+)
+
+// OverrideVisitor is visiting the changes during the override process
+// and allows to control what changes are allowed, or update the effective
+// value.
+//
+// For instance, it can disallow changes outside the specific path(s), or update
+// the location of the effective value.
+//
+// 'VisitDelete' is called when a value is removed from mapping or sequence
+// 'VisitInsert' is called when a new value is added to mapping or sequence
+// 'VisitUpdate' is called when a leaf value is updated
+type OverrideVisitor struct {
+	VisitDelete func(valuePath dyn.Path, left dyn.Value) error
+	VisitInsert func(valuePath dyn.Path, right dyn.Value) (dyn.Value, error)
+	VisitUpdate func(valuePath dyn.Path, left dyn.Value, right dyn.Value) (dyn.Value, error)
```

Comment:
Are there other use cases you have in mind besides updating the location of the result?

Reply:
I can't think of. Potentially, we can do better normalization, for instance, when two values mean the same, there should be no change.

Reply:
I just thought of a case: when we go back and forth into JSON, we'll have a type mismatch for numbers (ints vs floats), or time objects. I expect the kind mismatch to pick the right value over the left one, erasing the location. During normalization of the configuration tree the value will be converted back into what is expected for the path.

Reply:
Makes sense. Can we avoid this problem by normalizing output before passing it to override?

Reply:
You're right, that would fix it. That also surfaces issues with the returned payload (e.g. type mismatches).